solr/core/src/java/org/apache/solr/search/SolrQueryTimeoutImpl.java

```diff
@@ -67,8 +69,21 @@ public boolean shouldExit() {
   }
 
   /**
-   * Method to set the time at which the timeOut should happen.
-   * @param timeAllowed set the time at which this thread should timeout.
+   * Sets or clears the time allowed based on how much time remains from the start of the request plus the configured
+   * {@link CommonParams#TIME_ALLOWED}.
+   */
+  public static void set(SolrQueryRequest req) {
+    long timeAllowed = req.getParams().getLong(CommonParams.TIME_ALLOWED, -1L);
+    if (timeAllowed >= 0L) {
```

Comment:
Should be `>`, not `>=`. Doc on time allowed state that zero is no timeout, not immediate timeout. Looks like we were previously inconsistent about this.

Reply:
This seems inconsistent -- should we not be marking no timeout as -1?

Reply:
`>` vs `>=` is debatable; there's an argument both ways.  I suspect there's a test for it this way but moreover, I don't think we should change it.  It's fine.

Reply:
https://github.com/apache/lucene-solr/pull/1726/files#diff-d8beef800870f194d61993d701fd9cc2L77 has `>=`
https://github.com/apache/lucene-solr/pull/1726/files#diff-65e9f3712efc1ec962ea82a04a1d7aa1L104 has `>`

Either way, please update the docs at https://github.com/apache/lucene-solr/blob/master/solr/solrj/src/java/org/apache/solr/common/params/CommonParams.java#L162 because they are absolutely wrong (`>= 0` means no timeout???)

Reply:
MLT Handler was the exception it's used *way* less often than SearchHandler and so I ignored that.  I agree that the docs are wrong; I will update them.